api/core/v1alpha1/logset_types.go

```diff
@@ -87,11 +87,21 @@ func (l *LogSetBasic) GetStoreFailureTimeout() metav1.Duration {
 
 func (l *LogSetBasic) GetPVCRetentionPolicy() PVCRetentionPolicy {
 	if l.PVCRetentionPolicy == nil {
-		return PVCRetentionPolicyDelete
+		l.setDefaultRetentionPolicy()
 	}
 	return *l.PVCRetentionPolicy
 }
 
+func (l *LogSetBasic) GetS3RetentionPolicy() *PVCRetentionPolicy {
```

Comment:
How about add this method to the "S3" struct? So that we simply call `l.SharedStorage.S3.GetRetentionPolicy()`, which is automatically namespaced

Reply:
S3 retention policy is related to pvc retention policy, I think it is batter set in logsetbasic scope

Reply:
got it